src/python/zquantum/core/hamiltonian.py

```diff
@@ -69,3 +71,125 @@ def group_comeasureable_terms_greedy(
+    """Computes the variances of each frame in a grouped operator. 
+
+    If expectation values are provided, use variances from there, 
+    otherwise assume variances are 1 (upper bound). Correlation information
+    is ignored in the current implementation, covariances are assumed to be 0.
+
+    Args:
+        groups:  A list of QubitOperators that defines a (grouped) objective function
+        expecval: An ExpectationValues object containing the expectation
+            values of the operators and their squares. Optionally, contains
+            values of operator products to compute covariances.
+    Returns:
+        frame_variances: A Numpy array of the computed variances for each frame
+    """
+
+    if expecval is None:
+        frame_variances = [ np.sum(np.array(list(x.terms.values()))**2) for x in groups ] # Covariances are ignored; Variances are set to 1
+    else:
+        group_sizes = np.array([ len(g.terms.keys()) for g in groups ])
+        assert np.sum(group_sizes) == len(expecval.values)
+        real_expecval = expectation_values_to_real(expecval)
+        pauli_variances = 1. - real_expecval.values**2
+        frame_variances = []
+        for i, group  in enumerate(groups):
+            coeffs = np.array(list(group.terms.values()))
+            offset = 0 if i == 0 else np.sum(group_sizes[:i])
+            pauli_variances_for_group = pauli_variances[offset : offset + group_sizes[i]]
+            frame_variances.append(np.sum(coeffs**2 * pauli_variances_for_group))
+
+    return np.array(frame_variances)
+
+def get_expectation_values_from_rdms(interactionrdm: InteractionRDM, qubitoperator: QubitOperator, 
+    sort_terms: bool = False) -> ExpectationValues:
+    """Computes expectation values of Pauli strings in a QubitOperator given a fermionic InteractionRDM from
+       OpenFermion.
+
+    Args:
+        interactionrdm: interaction RDM to use for the expectation values
+            computation, as an OF InteractionRDM object
+        qubitoperator: qubit operator to compute the expectation values for
+            in the form of an OF QubittOperator object
+        sort_terms: whether or not the input qubit operator needs to be sorted before calculating expectations
+    Returns:
+        expectation values of Pauli strings in the qubit operator as an ExpectationValues object
+    """
+    if sort_terms:
+        terms_iterator = sorted(
+            qubitoperator.terms.items(), key=lambda x: abs(x[1]), reverse=True
+        )
+    else:
+        terms_iterator = qubitoperator.terms.items()
+    reordered_qubitoperator = QubitOperator()
+    for term, coefficient in terms_iterator:
+        reordered_qubitoperator += QubitOperator(term, coefficient)
+
+    expectations_packed = interactionrdm.get_qubit_expectations(reordered_qubitoperator)
+
+    if () in expectations_packed.terms:
+        del expectations_packed.terms[()] # Expectation of the constant term is excluded from expectation values
```

Comment:
I have eliminated the constant term in `compute_group_variances` as well. The same convention was originally adopted in `get_estimated_expectation_values` for the estimators. It does get appended at the end after concatenating all the measurement sets but I think it's not a very good design choice since we don't have a way to tell whether we included it or not without knowing the initial qubit operator. I'd be curious to know what @max-radin thinks.